lntest/itest/lnd_recovery_test.go

```diff
@@ -282,3 +289,156 @@ func testOnchainFundRecovery(ht *lntemp.HarnessTest) {
+// addresses the following scenario:
+//	1. A key is derived from the internal m/1017' scope with a custom key
+//	   family and a p2wkh address is derived from that key.
+//	2. Funds are sent to the address created above in a way that also
+//	   creates a change output. The change output is recognized as belonging
+//	   to the wallet, which is correct.
+//	3. The funds on the address created in step 1 are fully spent (without
+//	   creating a change output) into an output that doesn't belong to the
+//	   wallet (e.g. a channel funding output).
+//	4. At some point the user re-scans their wallet by using the
+//	   --reset-wallet-transactions flag.
+//	5. The wallet re-scan detects the change output created in step 2 and
+//	   flags the transaction as relevant.
+//	6. While adding the relevant TX to the wallet DB, the wallet also
+//	   detects the address from step 1 as belonging to the wallet (because
+//	   the internal key scope is defined as having the address type p2wkh)
+//	   and adds that output as an UTXO as well (<- this is the bug). The
+//	   wallet now has two UTXOs in its database.
+//	7. The transaction that spends the UTXO of the address from step 1 is
+//	   not detected by the wallet as belonging to it (because the output is
+//	   a channel output and the input (correctly) isn't recognized as
+//	   belonging to the wallet in that part of the code, it is never marked
+//	   as spent and stays in the wallet as a ghost UTXO forever.
+// The fix in the wallet is simple: In step 6, don't detect addresses from
+// internal scopes while re-scanning to be in line with the logic in other areas
+// of the wallet code.
+func testRescanAddressDetection(ht *lntemp.HarnessTest) {
+	// We start off by creating a new node with the wallet re-scan flag
+	// enabled. This won't have any effect on the first startup but will
+	// come into effect after we re-start the node.
+	walletPassword := []byte("some-password")
+	carol, _, _ := ht.NewNodeWithSeed(
+		"carol", []string{"--reset-wallet-transactions"},
+		walletPassword, false,
+	)
+	ht.FundCoins(btcutil.SatoshiPerBitcoin, carol)
+
+	// Create an address generated from internal keys.
+	keyDesc := carol.RPC.DeriveNextKey(&walletrpc.KeyReq{KeyFamily: 123})
+	pubKeyHash := btcutil.Hash160(keyDesc.RawKeyBytes)
+	ghostUtxoAddr, err := btcutil.NewAddressWitnessPubKeyHash(
+		pubKeyHash, harnessNetParams,
+	)
+	require.NoError(ht, err)
+
+	// Send funds to the (p2wkh!) address generated from the internal
+	// (m/1017') key scope. Because the internal key scope is defined as
+	// p2wkh address type, this might be incorrectly detected by the wallet
+	// in some situations (which this test makes sure is fixed).
+	const ghostUtxoAmount = 456_000
+	carol.RPC.SendCoins(&lnrpc.SendCoinsRequest{
+		Addr:        ghostUtxoAddr.String(),
+		Amount:      ghostUtxoAmount,
+		SatPerVbyte: 1,
+	})
+	ht.Miner.AssertNumTxsInMempool(1)
```

Comment:
nit: could use `ht.MineBlocksAndAssertNumTxes(1)` instead, which checks the mempool and mine blocks.

Reply:
Ah, wasn't aware that existed, but good to know. Changed it to that call.